analytics.py

```diff
@@ -30,5 +37,5 @@ def __init__(self, _db):
     def collect_event(self, library, license_pool, event_type, time=None, **kwargs):
         if not time:
             time = datetime.datetime.utcnow()
-        for provider in self.providers:
+        for provider in (self.sitewide_providers + self.library_providers[library.id]):
```

Comment:
Adding library ID to CirculationEvents makes a lot of sense. File a bug for that and then :+1: